e2e/run.js

```diff
@@ -0,0 +1,113 @@
+const { spawn, spawnSync } = require('child_process')
+
+const debug = false
+const createPostgresContainer = true
+const deletePostgresContainer = true
+
+const postgresContainerName = 'indexer-postgres-e2e'
+
+function run(command, args, options) {
+  options = options || {}
+  if (debug) console.log(`${command} \\\n  ${args.join(' \\\n  ')}`)
+  const result = spawnSync(command, args, options)
+  if (result.error) {
+    throw result.error
+  } else if (result.status) {
+    throw new Error(`${command} failed: ${result.status}`)
+  } else if (result.signal) {
+    throw new Error(`${command} exited from signal: ${result.signal}`)
+  }
+  return result
+}
+
+function docker(args, options) {
+  options = options || {}
+  options = {
+    ...options,
+    env: {
+      ...process.env,
+      ...options.env
+    }
+  }
+  return run('docker', args, options)
+}
+
+async function psql(statement, timeoutSeconds) {
```

Comment:
This function looks a bit complex. I think it can probably be simplified if instead of querying using the command-line docker we just access the database the same way the indexer does using this file https://github.com/valora-inc/indexer/blob/main/src/database/db.ts

Reply:
I think a potential problem is that the db has not been initialized yet at this point.

One alternative is to have a separate setup script that creates the container/database and in this place we just run data setup (mostly deleting everything, in this case it would be a `DELETE FROM table_name;` for each table).

That would also make running locally easier/faster I think because we wouldn't need to recreate the container each time and the DELETE operations should run fast since we shouldn't be creating too many records for e2e tests.

What do you think?

Reply:
good point on using the client library -- I can do that.

Why do you want to avoid recreating the container?

Reply:
...and to be clear, we should use the postgres client bindings, but avoid using knex or applying the migrations (since that would essentially be testing some other amalgamation that we don't actually deploy)

Reply:
I tried running it locally and got an error `Error: docker failed: 125` :/ 

I guess the reason is that I wonder how quickly the container is created. I was thinking about the development cycle speed when adding new tests. For running in CI it's the same thing I guess but I'm not sure if my worries about the time to create the container are valid or not. If it's a quick thing then recreating sounds great

Reply:
> ...and to be clear, we should use the postgres client bindings, but avoid using knex or applying the migrations (since that would essentially be testing some other amalgamation that we don't actually deploy)

I'm not sure I follow, knex is just a library to interact with pg, can we not use it (without running the migrations) to query? It would turn this big function into just 
```
  const result = await database.raw(sql)
  return result.rows
```

Reply:
I guess this current way using the pg bindings is simple enough, nevermind.

I would move it to another file though, just to separate the db interaction from the rest of the e2e setup.

Reply:
Moved!